packages/frontend/Dockerfile

```diff
@@ -8,48 +8,16 @@ ENV PATH="$PNPM_HOME:$PATH"
 RUN corepack enable
 RUN corepack prepare pnpm@8.7.4 --activate
 
-COPY pnpm-lock.yaml ./
+COPY pnpm-lock.yaml package.json pnpm-workspace.yaml .npmrc tsconfig.json tsconfig.build.json ./
 
 RUN --mount=type=cache,id=pnpm,target=/pnpm/store \
     pnpm fetch \
     | grep -v "cross-device link not permitted\|Falling back to copying packages from store"
 
-FROM base AS prod-deps
```

Comment:
why do we need to remove the multi-step builds here?

Reply:
We are running the admin UI container in dev mode now so we need the dev dependencies not the prod dependencies. 

This Dockerfile now uses only a builder stage basically, because we're mounting code in, which we still need to build and run inside the container. This is so that we can see live changes with a  simple refresh instead of rebuilding the container container each time we want to see our changes reflect.

Reply:
and this is required because we cant do hot module reloading when launching with `remix-serve` (even with the bind mount), is that correct?

Reply:
is it a bad practice to have separate docker files for dev/prod? I see some mixed conclusions on the subject ([1](https://stackoverflow.com/questions/46440909/how-to-configure-different-dockerfile-for-development-and-production), [2](https://www.reddit.com/r/kubernetes/comments/ukhi3t/different_dockerfile_for_dev_vs_stagingprod/), [3](https://stackoverflow.com/questions/48116279/dockerfile-setup-for-both-production-and-development)).

Just thinking we'll have no reason to run it locally so errors might go unnoticed. Thinking in particular with remix upgrades. I suppose at least the image build would be verified by CI.

I guess HMR necessitates different run commands and dev deps but I wonder if we could unify it a bit more up to that point. Perhaps using multi-stage builds and targets like suggested [here](https://stackoverflow.com/a/56114663/12893676) or perhaps controlling with 
[ARG](https://docs.docker.com/reference/dockerfile/#arg) (as some of the above links have alluded to).

Reply:
in any case, can we install the dev dependencies in the dockerfile instead of in the container after start? And we can move the dev command to the dockerfile as well I think, instead of this in the compose: `command: sh -c "pnpm install && pnpm dev"`

Reply:
Thanks for that catch. I had it that way for a previous configuration. That additional `pnpm install` is no longer needed in the docker-compose. 

As for your concern about having a production use Dockerfile that we're not forced to interact with, or even worse, can't easily interact with... That is a valid problem...

Reply:
would need to install dev dependencies too so we can use the `remix dev` cmd